hibernate-reactive-core/src/test/java/org/hibernate/reactive/ReactiveStatelessUpdateTest.java

```diff
@@ -0,0 +1,101 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive;
+
+import java.io.Serializable;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+import org.hibernate.cfg.Configuration;
+
+import org.junit.After;
+import org.junit.Test;
+
+import io.vertx.ext.unit.TestContext;
+
+public class ReactiveStatelessUpdateTest  extends BaseReactiveTest {
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+		configuration.addAnnotatedClass( SampleEntity.class );
+		configuration.addAnnotatedClass( SampleJoinEntity.class );
+		return configuration;
+	}
+
+	@After
+	public void cleanDb(TestContext context) {
+		test( context, deleteEntities( "SampleEntity", "SampleJoinEntity" ) );
+	}
+
+	@Test
+	public void testUpdateWithDetachedEntity(TestContext context) {
+		SampleEntity sampleEntity = new SampleEntity();
+		sampleEntity.sampleField = "test";
+
+		SampleJoinEntity sampleJoinEntity = new SampleJoinEntity();
+		sampleJoinEntity.sampleEntity = sampleEntity;
+
+		test( context, getMutinySessionFactory()
+				.withTransaction( (s, t) -> s.persistAll( sampleEntity, sampleJoinEntity ) )
+
+				.chain( targetId -> getMutinySessionFactory()
+						.withTransaction( (session, transaction) -> session
+								.find( SampleJoinEntity.class, sampleJoinEntity.id ) )
+				)
+
+				.chain( joinEntityFromDatabase -> getMutinySessionFactory().withStatelessTransaction(
+						(s, t) -> {
+							SampleEntity entityFromDb = joinEntityFromDatabase.sampleEntity;
+							entityFromDb.sampleField = "updated field";
+							// Without the fix it fails here
+							return s.update( entityFromDb );
+						} )
+				)
```

Comment:
I would add a query to check that the value has been actually updated

Reply:
I'll add it